internal/scanner/scanner.go

```diff
@@ -0,0 +1,88 @@
+package scanner
+
+import (
+	"fmt"
+	"unicode/utf8"
+
+	"github.com/influxdata/flux/internal/token"
+)
+
+//go:generate ruby unicode2ragel.rb -e utf8 -o unicode.rl
+//go:generate ragel -I. -Z scanner.rl -o scanner.gen.go
+//go:generate sh -c "go fmt scanner.gen.go > /dev/null"
```

Comment:
These generate commands now mean that both ragel and ruby are dependencies of the Flux repo. Do we have that documented anywhere? Is there a specific version of ragel that we need to use?

Reply:
I was planning to rewrite the ruby one in the future to use go. There's no specific version of ragel that I'm aware of. I don't think they have had a release for a long time.

I'll add a note that they're dependencies until I get to making them optional.

Reply:
Ragel 6 is still the stable supported release.  Ragel 7 is in development (with a slightly broken Go backend.)